Dockerfiles/midstream/entrypoint.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -e
+main() {
+    local username="${SSH_AUTH_USER:-default}"
+    local home
+    home=$(mk_home "$username")
+    setup_passwd "$username" "$home"
+    exec sleep inf
+}
+mk_home() {
+    local username="${1:?}"
+    [[ -d /workDir ]] && [[ -w /workDir ]] && \
+        mkdir -p "/workDir/home/$username" && \
+        echo "/workDir/home/$username" && \
+        return 0
+    mktemp --tmpdir -d "${username}.XXXXXX"
+}
+setup_passwd() {
+    local username="${1:?}"
+    local home="${2:?}"
+    local uid="$3"
+    local gid="${4:-0}"
+    local passwd=/etc/passwd
+    ! whoami 2>/dev/null || return 0
```

Comment:
We are interested in entry in /etc/passwd. Might be better to check it directly.
```
getent passwd $uid
```

Reply:
Resolved